pillarbox-player/src/main/java/ch/srgssr/pillarbox/player/qos/QoSCoordinator.kt

```diff
@@ -76,16 +99,17 @@ internal class QoSCoordinator(
         session: PlaybackSessionManager.Session,
         data: Any? = null,
     ) {
-        val message = QoSMessage(
-            data = data ?: metricsCollector.getCurrentMetrics().toQoSEvent(),
-            eventName = eventName,
-            sessionId = session.sessionId,
-        )
-
-        messageHandler.sendEvent(message)
+        metricsCollector.getCurrentMetrics()?.let {
```

Comment:
Is it fine to not send the event if `data` is not `null` but `metricsCollector.getCurrentMetrics()` is?
Maybe the `data` parameter of `QoSMessage` should be extracted in a variable:
```kotlin
val dataToSend = data ?: metricsCollector.getCurrentMetrics()?.toQoSEvent() ?: return
```